docs/DeveloperGuide.md

```diff
@@ -1,38 +1,228 @@
 # Developer Guide
 
+## Table of Contents
++ [Acknowledgements](#acknowledgements)
++ [Design](#design)
+   * [Architecture](#architecture)
+   * [UI Component](#ui-component)
+   * [Storage Component](#storage-component)
++ [Implementation](#implementation)
++ [Product Scope](#product-scope)
+   * [Target User Profile](#target-user-profile)
+   * [Value Proposition](#value-proposition)
++ [User Stories](#user-stories)
++ [Non-Functional Requirements](#non-functional-requirements)
++ [Glossary](#glossary)
++ [Instructions for Manual Testing](#instructions-for-manual-testing)
+
+
+
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+## Design 
+### Architecture
+
+### Parser Component
+The `Parser` class handles the parsing of user input, executing the appropriate tasks
+or throwing relevant errors.
+* `Parser` first receives user input from `Duke` through `Parser::handleCmd`
+* Depending on the command given, `Parser` will then execute the command on the `ExpenseManager`instance, or throw a `DukeException`, which will be caught by `Duke`
+### **Sequence Diagram**
+<img alt="parser sequence diagram" src="diagrams/parser.png" width="280" />
+
+### UI Component
+The API of this component is specified in Ui.java \
+The Ui class contains several static methods which display messages to the user when the user interacts with program.\
+Types of messages displayed:
++ Instructions: Information message that describes the list of all commands of the application.
++ Greeting Messages: Messages shown to greet the user and acknowledge their exit.
++ Error Messages: Messages shown to inform the user that their input is invalid.
++ Category Choice: Displays category choices of expense to the user. \
+The class also prints text such as lines and the formatted list.
+  
+<img alt="ui class diagram" src="diagrams/UiComponentClassDiagram.png" width="280" />
+
+### Storage Component
+
+<img alt="storage sequence diagram" src="diagrams/storage.png" width="280" />
```

Comment:
![image](https://user-images.githubusercontent.com/88707726/228733401-d3d76729-a690-4c21-9d71-2ddb840b406e.png)
missing semicolon in front of each class